pandablocks/utils.py

```diff
@@ -60,11 +60,18 @@ def words_to_table(
         if field_info.subtype == "int":
             # First convert from 2's complement to offset, then add in offset.
             packing_value = (value ^ (1 << (bit_length - 1))) + (-1 << (bit_length - 1))
+            packing_value = packing_value.astype(np.int32)
         elif field_info.subtype == "enum" and convert_enum_indices:
             assert field_info.labels, f"Enum field {field_name} has no labels"
             packing_value = [field_info.labels[x] for x in value]
         else:
-            packing_value = value
+            # Use shorter types, as these are used in waveform creation
```

Comment:
The `UnpackedArray` type needs updating with the new possible returns.

It may also be good to add the types to both `value` and `packing_value` e.g.:
```
value: np.uint32 = (packed[word_offset] >> bit_offset) & mask

packing_value: UnpackedArray
```

Reply:
Type hinting `UnpackedArray` for these doesn't work because mypy doesn't understand when `List[str]` is or isn't a possibility.

Reply:
Apologies, I wrote an incorrect type hint for `value`. When I write the following lines I see no type errors anymore:

```
UnpackedArray = Union[
    npt.NDArray[np.int32],
    npt.NDArray[np.uint8],
    npt.NDArray[np.uint16],
    npt.NDArray[np.uint32],
    List[str],
]



...

value: npt.NDArray[np.uint32] = (packed[word_offset] >> bit_offset) & mask

packing_value: UnpackedArray

...
if field_info.subtype == "int":
            # First convert from 2's complement to offset, then add in offset.
            temp = (value ^ (1 << (bit_length - 1))) + (-1 << (bit_length - 1))
            packing_value = temp.astype(np.int32)

```